internal/test/framework/framework.go

```diff
@@ -369,6 +395,100 @@ func (f *TestFramework) RetrieveArtifacts() {
 	}
 }
 
+// ApplyHybridOverlayPatch will enable the hybrid overlay on the cluster
+func (f *TestFramework) ApplyHybridOverlayPatch() error {
+	jsonPatch := []byte(`{"spec":{"defaultNetwork":{"ovnKubernetesConfig":{"hybridOverlayConfig":` +
+		`{"hybridClusterNetwork":[{"cidr":"10.132.0.0/14","hostPrefix":23}]}}}}}`)
+
+	_, err := f.OSOperatorClient.Networks().Patch("cluster", k8stypes.MergePatchType, jsonPatch)
+	if err != nil {
+		return fmt.Errorf("could not apply patch %s to cluster network: %s", string(jsonPatch), err)
+	}
+
+	// Wait until the hybrid overlay changes are complete
+	return f.waitUntilHybridOverlayReady()
+}
+
+// waitUntilHybridOverlayReady returns once OVN is ready again, after applying the hybrid overlay patch.
+func (f *TestFramework) waitUntilHybridOverlayReady() error {
+	err := f.waitUntilMasterNodesAnnotated()
+	if err != nil {
+		return err
+	}
+
+	// This is being done after we wait for the master nodes, as we need to make sure the patch is being acted on,
+	// and we are not checking the pods before they begin to restart with the hybrid overlay changes
+	err = f.waitUntilOVNPodsReady()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// waitUntilOVNPodsReady returns when either all pods in the openshift-ovn-kubernetes namespace are ready, or the
+// timeout limit has been reached.
+func (f *TestFramework) waitUntilOVNPodsReady() error {
+	for i := 0; i < RetryCount; i++ {
+		pods, err := f.K8sclientset.CoreV1().Pods("openshift-ovn-kubernetes").List(metav1.ListOptions{})
+		if err != nil {
+			return fmt.Errorf("could not get pods: %s", err)
+		}
+
+		allReady := true
+		for _, pod := range pods.Items {
+			podReady := false
+			for _, condition := range pod.Status.Conditions {
+				if condition.Type == v1.PodReady {
+					podReady = true
+					break
+				}
+			}
+			if !podReady {
+				allReady = false
+				break
+			}
+		}
+		if allReady {
+			return nil
+		}
+		time.Sleep(RetryInterval)
+	}
+	return fmt.Errorf("timed out waiting for pods in namespace \"openshift-ovn-kubernetes\" to be ready")
+
+}
+
+// waitUntilMasterNodesAnnotated returns when either all master nodes have had the hybrid overly gateway mac annotation
+// applied to them, or reaches a timeout limit.
+func (f *TestFramework) waitUntilMasterNodesAnnotated() error {
+	masterLabel := "node-role.kubernetes.io/master"
```

Comment:
I misspoke about it being just master nodes. These annotations are applied on both `master` and `worker` nodes so they all need to be checked.